modules/api/src/main/java/org/eclipse/xpanse/api/controller/EndUserQueryController.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.api.controller;
+
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.annotation.Resource;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.modules.database.register.RegisterServiceEntity;
+import org.eclipse.xpanse.modules.models.enums.Category;
+import org.eclipse.xpanse.modules.models.enums.Csp;
+import org.eclipse.xpanse.modules.models.query.RegisteredServiceQuery;
+import org.eclipse.xpanse.modules.models.view.CategoryOclVo;
+import org.eclipse.xpanse.modules.models.view.UserAvailableServiceVo;
+import org.eclipse.xpanse.orchestrator.register.RegisterService;
+import org.springframework.beans.BeanUtils;
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+
+/**
+ * REST interface methods for end-user query available managed services.
+ */
+@Slf4j
+@RestController
+@RequestMapping("/xpanse/enduser")
+@CrossOrigin
+public class EndUserQueryController {
+
+    @Resource
+    private RegisterService registerService;
+
+    /**
+     * List the available managed services.
+     *
+     * @param categoryName   name of category.
+     * @param cspName        name of cloud service provider.
+     * @param serviceName    name of registered service.
+     * @param serviceVersion version of registered service.
+     * @return response
+     */
+    @Tag(name = "Available Managed Services",
+            description = "APIs to query the available managed services.")
+    @Operation(description = "List the available managed services.")
+    @GetMapping(value = "/register",
```

Comment:
```suggestion
    @GetMapping(value = "/services/available",
```
Further to our previous discussions, we should rename this one to "/services/available".

Reply:
I  rename these end-user APIs  with presuffix route "/xpanse/services/available".